internal/tools/aws/database_migration.go

```diff
@@ -0,0 +1,132 @@
+func (d *RDSDatabaseMigration) Setup(logger log.FieldLogger) (string, error) {
+	masterInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMasterInstanceID(d.masterInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	slaveInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMigrationInstanceID(d.slaveInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = d.awsClient.Service().ec2.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: masterInstanceSG.GroupId,
+		IpPermissions: []*ec2.IpPermission{
+			{
+				FromPort:   aws.Int64(3306),
+				IpProtocol: aws.String("tcp"),
+				ToPort:     aws.Int64(3306),
+				UserIdGroupPairs: []*ec2.UserIdGroupPair{
+					{
+						Description: aws.String("Ingress Traffic from other RDS instance"),
+						GroupId:     slaveInstanceSG.GroupId,
+					},
+				},
+			},
+		},
+	})
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return "", err
+	}
+
+	logger.WithField("migration-installation-id", d.masterInstallationID).Info("database migration setup completed")
+
+	return model.DatabaseMigrationStatusSetupComplete, nil
+}
+
+// Teardown removes access from one RDS database to another and rollback any previous database configuration.
+func (d *RDSDatabaseMigration) Teardown(logger log.FieldLogger) (string, error) {
+	masterInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMasterInstanceID(d.masterInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	slaveInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMigrationInstanceID(d.slaveInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = d.awsClient.Service().ec2.RevokeSecurityGroupIngress(&ec2.RevokeSecurityGroupIngressInput{
+		GroupId: masterInstanceSG.GroupId,
+		IpPermissions: []*ec2.IpPermission{
+			{
+				FromPort:   aws.Int64(3306),
+				IpProtocol: aws.String("tcp"),
+				ToPort:     aws.Int64(3306),
+				UserIdGroupPairs: []*ec2.UserIdGroupPair{
+					{
+						GroupId: slaveInstanceSG.GroupId,
+					},
+				},
+			},
+		},
+	})
+	if err != nil && !strings.Contains(err.Error(), "rule does not exist") {
+		return "", err
+	}
+
+	logger.WithField("migration-installation-id", d.masterInstallationID).Info("database migration teardown completed")
+
+	return model.DatabaseMigrationStatusTeardownComplete, nil
+}
+
+// Replicate starts the process for replicating an master RDS database. This method must return an resplication status or an error.
+func (d *RDSDatabaseMigration) Replicate(logger log.FieldLogger) (string, error) {
+	return "", errors.New("not implemented")
+}
+
+func (d *RDSDatabaseMigration) describeDBInstanceSecurityGroup(instanceID string) (*ec2.SecurityGroup, error) {
+	output, err := d.awsClient.Service().rds.DescribeDBInstances(&rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: aws.String(instanceID),
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Just a reminder to be sure to `Wrap` these bare `err`s with some more context when you `return` them, before merge. I know there can be a lot of churn with these statements during review so I assume you intend to do this before merge and just haven't yet.